tables/AlternateMans/StokesIStMan.h

```diff
@@ -0,0 +1,154 @@
+#ifndef CASACORE_STOKES_I_STORAGE_MANAGER_H_
+#define CASACORE_STOKES_I_STORAGE_MANAGER_H_
+
+#include <casacore/tables/DataMan/DataManager.h>
+
+#include <casacore/casa/Containers/Record.h>
+
+#include "BufferedColumnarFile.h"
+
+#ifndef DOXYGEN_SHOULD_SKIP_THIS
+extern "C" {
+#endif
+void register_stokesistman();
+#ifndef DOXYGEN_SHOULD_SKIP_THIS
+}
+#endif
+
+namespace casacore {
+
+class StokesIStManColumn;
+
+/**
+ * The Stokes I storage manager behaves like a full set of (4) polarizations but
+ * only stores the Stokes I value on disk.
+ */
+class StokesIStMan final : public casacore::DataManager {
+ public:
+  /**
+   * This constructor is called by Casacore when it needs to create a
+   * StokesIStMan. When Casacore loads an StokesIStMan for an existing MS, the
+   * "spec" parameter will be empty, thus the class should initialize its
+   * properties by reading them from the file. The @p spec is used to make a new
+   * storage manager with specs similar to another one.
+   * @param name Name of this storage manager.
+   * @param spec Specs to initialize this class with.
+   */
+  StokesIStMan(const casacore::String &name, const casacore::Record &spec);
+
+  /**
+   * Copy constructor that initializes a storage manager with similar specs.
+   * The columns are not copied: the new manager will be empty.
+   */
+  StokesIStMan(const StokesIStMan &source);
+
+  ~StokesIStMan() noexcept;
+
+  /**
+   * This StokesIStMan takes the settings of the source (but not the
+   * columns and/or data).
+   */
+  StokesIStMan &operator=(const StokesIStMan &source) = delete;
+
+  casacore::DataManager *clone() const final { return new StokesIStMan(*this); }
+  /**
+   * Create an object with given name and spec.
+   * This methods gets registered in the DataManager "constructor" map.
+   * The caller has to delete the object. New class will be
+   * initialized via @ref DyscoStMan(const casacore::String& name, const
+   * casacore::Record& spec).
+   * @returns A DyscoStMan with given specs.
+   */
+  static casacore::DataManager *makeObject(const casacore::String &name,
+                                           const casacore::Record &spec) {
+    return new StokesIStMan(name, spec);
+  }
+
+  casacore::String dataManagerType() const final { return "StokesIStMan"; }
+
+  casacore::Record dataManagerSpec() const final;
+
+  casacore::Bool canAddRow() const final { return true; }
+
+  casacore::Bool canRemoveRow() const final { return true; }
```

Comment:
Would it make sense to put this to `False`? If I understand correctly, it can only remove rows at the end, not sure how big the use case for that is.